plugins/entity_filter.py

```diff
@@ -0,0 +1,432 @@
+        )
+
+
+class EntityFilter(QHAnaPluginBase):
+
+    name = _plugin_name
+    version = __version__
+
+    def __init__(self, app: Optional[Flask]) -> None:
+        super().__init__(app)
+
+    def get_api_blueprint(self):
+        return ENTITY_FILTER_BLP
+    
+    def get_requirements(self) -> str:
+        return ""
+
+
+TASK_LOGGER = get_task_logger(__name__)
+
+
+@CELERY.task(name=f"{EntityFilter.instance.identifier}.entity_filter_task", bind=True)
+def entity_filter_task(self, db_id: int) -> str:
+    TASK_LOGGER.info(f"Starting new entity filter task with db id '{db_id}'")
+    task_data: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+
+    if task_data is None:
+        msg = f"Could not load task data with id {db_id} to read parameters!"
+        TASK_LOGGER.error(msg)
+        raise KeyError(msg)
+
+    params : Dict = loads(task_data.parameters or "{}")
+    input_file_url: Optional[str] = params.get("input_file_url", None)
+    attributes: Optional[str] = params.get("attributes", None)
+    attributes_setting: Optional[str] = params.get("attributes_setting", None)
+    n_rows: Optional[int] = params.get("n_rows", None)
+    row_sampling : Optional[str] = params.get("row_sampling", None)
+    id_list: Optional[str] = params.get("id_list", None)
+    
+    TASK_LOGGER.info(f"Loaded input parameters from db: input_file_url='{input_file_url}', \
+        attributes='{attributes}', attributes_setting='{attributes_setting}', n_rows='{n_rows}', \
+        row_sampling='{row_sampling}', id_list='{id_list}'")
+
+    if input_file_url is None or not input_file_url:
+        raise ValueError("No input file URL provided!") 
+   
+    if not attributes: # empty string or None
+        attributes = []
+        
+    if n_rows is None and not id_list:
+        raise ValueError("Row number argument not provided!")
+    if not id_list:
+        id_list = []
+        n_sampled_rows = n_rows
+    else:
+        if n_rows is None:
+            n_rows = len(id_list)
+        n_sampled_rows = max(0, n_rows - len(id_list)) 
+
+    if len(id_list) > n_rows:
+        raise ValueError("Length of ID list greater than number of rows!")
+
+    if row_sampling is None and n_sampled_rows > 0:
+        raise ValueError("Row sampling not specified!")
+
+    if attributes_setting is None:
+        raise ValueError("Attribute setting not specified!")
+    
+    # Filter rows: get requested entities from input file 
+    output_entities_id_list = []
+    # Random sampling as in https://math.stackexchange.com/questions/846036/can-i-uniformly-sample-from-n-distinct-elements-where-n-is-unknown-but-fini
+    output_entities_random_rows = []
+    sampling_counter = 0
+    mimetype = mimetypes.MimeTypes().guess_type(input_file_url)[0]
+    with open_url(input_file_url, stream=True) as url_data:
+        for entity in ensure_dict(load_entities(file_=url_data, mimetype=mimetype)):
```

Comment:
For the local file used for testing url_data.headers only contains 'Content-Length'. Another option would be to try 
```suggestion
        try:
            mimetype = url_data.headers['Content-Type']
        except:
            mimetype = mimetypes.MimeTypes().guess_type(input_file_url)[0]
```
or use python-magic.

Reply:
For local files the `FileAdapter` in `qhana_plugin_runner/util/request_helpers.py` can be extended to directly guess the mimetype of the file/path. This would not replace the proposed try-except, but the except clause should only listen for `KeyError`!